api/rest/client/methods_test.go

```diff
@@ -308,3 +317,115 @@ func TestGetConnectGraph(t *testing.T) {
 
 	testClients(t, api, testF)
 }
+
+type waitService struct {
+	l        sync.Mutex
+	pinStart time.Time
+}
+
+func (wait *waitService) Pin(ctx context.Context, in api.PinSerial, out *struct{}) error {
+	wait.l.Lock()
+	defer wait.l.Unlock()
+	wait.pinStart = time.Now()
+	return nil
+}
+
+func (wait *waitService) Status(ctx context.Context, in api.PinSerial, out *api.GlobalPinInfoSerial) error {
+	wait.l.Lock()
+	defer wait.l.Unlock()
+	c1, _ := cid.Decode(in.Cid)
+	if time.Now().After(wait.pinStart.Add(5 * time.Second)) { //pinned
+		*out = api.GlobalPinInfo{
+			Cid: c1,
+			PeerMap: map[peer.ID]api.PinInfo{
+				test.TestPeerID1: {
+					Cid:    c1,
+					Peer:   test.TestPeerID1,
+					Status: api.TrackerStatusPinned,
+					TS:     wait.pinStart,
+				},
+				test.TestPeerID2: {
+					Cid:    c1,
+					Peer:   test.TestPeerID2,
+					Status: api.TrackerStatusPinned,
+					TS:     wait.pinStart,
+				},
+			},
+		}.ToSerial()
+	} else { // pinning
+		*out = api.GlobalPinInfo{
+			Cid: c1,
+			PeerMap: map[peer.ID]api.PinInfo{
+				test.TestPeerID1: {
+					Cid:    c1,
+					Peer:   test.TestPeerID1,
+					Status: api.TrackerStatusPinning,
+					TS:     wait.pinStart,
+				},
+				test.TestPeerID2: {
+					Cid:    c1,
+					Peer:   test.TestPeerID2,
+					Status: api.TrackerStatusPinned,
+					TS:     wait.pinStart,
+				},
+			},
+		}.ToSerial()
+	}
+
+	return nil
+}
+
+func TestWaitFor(t *testing.T) {
+	tapi := testAPI(t)
+	defer shutdown(tapi)
+
+	rpcS := rpc.NewServer(nil, "wait")
+	rpcC := rpc.NewClientWithServer(nil, "wait", rpcS)
+	err := rpcS.RegisterName("Cluster", &waitService{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tapi.SetClient(rpcC)
+
+	testF := func(t *testing.T, c *Client) {
+		ci, _ := cid.Decode(test.SlowCid)
```

Comment:
`test.SlowCid` doesn't exist. Related question, where do the `test.TestCid1`, etc, come from when running the tests?